woocommerce/woocommerce
# Fix e2e tests required job being skipped on failures

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #44313 .

With changes introduced by #43056 the `Run e2e tests.` job is skipped if there's a test failure. This job is required in GitHub checks, but if it's skipped it's getting ignored, failing its purpose.

The fix is to always run the job, evaluate the results of the e2e jobs is depends on and fail or pass accordingly.

I also renamed the job to `Evaluate e2e tests results` to better represent the action. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Break an e2e test and push the change to trigger the tests job.
2. Check the `Run tests against PR / Evaluate e2e tests results` job. It should run, and it should be failed. Validated with [this run](https://github.com/woocommerce/woocommerce/actions/runs/7756967057/workflow?pr=44308). (before renaming the job).
4. Check the `Run tests against PR in an environment with HPOS disabled / Evaluate e2e tests results` job. It should run, and it should be failed. Validated with [this run](https://github.com/woocommerce/woocommerce/actions/runs/7756967042/job/21155435521?pr=44308) (before renaming the job).
5. Fix the broken test and run the tests again.
6. Check the `Run tests against PR / Evaluate e2e tests results` job. It should run, and it should be successful. Validated with [this run](https://github.com/woocommerce/woocommerce/actions/runs/7758579357/job/21161430605?pr=44308).
7. Check the `Run tests against PR in an environment with HPOS disabled / Evaluate e2e tests results` job. It should run, and it should be successful. Validated with [this run](https://github.com/woocommerce/woocommerce/actions/runs/7758579335/job/21161411686?pr=44308).




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
